src/containers/NahmiiWithdraw/tests/index.test.js

```diff
@@ -0,0 +1,452 @@
+            expect(wrapper.find('.start-settlement .challenge-btn').props().disabled).toEqual(false);
+          });
+          it('should disable settle button when required staged amount is greater than nahmii balance', () => {
+            const wrapper = shallow(
+              <NahmiiWithdraw
+                {...props}
+                {...getProps(t)}
+                currentWalletWithInfo={
+                  getProps(t).currentWalletWithInfo.setIn(
+                    ['balances', 'nahmiiCombined', 'assets'],
+                    fromJS([{ balance: new BigNumber(1), symbol: 'ETH', currency: 'ETH' }])
+                  ).set('type', 'software')
+                }
+              />
+            );
+            wrapper.setState({ amountToWithdraw: new BigNumber(2) });
+            expect(wrapper.find('.start-settlement .nahmii-balance-before-staging').props().main).toEqual('1 ETH');
+            expect(wrapper.find('.start-settlement .nahmii-balance-after-staging').props().main).toEqual('-1 ETH');
+            expect(wrapper.find('.start-settlement .challenge-btn').props().disabled).toEqual(true);
+          });
+        });
+      });
+
+      it('should clear withdraw input when status is success', () => {
+        const wrapper = shallow(
+          <NahmiiWithdraw
+            {...props}
+          />
+        );
+
+        wrapper.setState({
+          amountToWithdrawInput: '1',
+          amountToWithdraw: new BigNumber('1'),
+        });
+        expect(wrapper.find('.withdraw-input').props().value).toEqual('1');
+        wrapper.setProps({
+          ongoingChallenges: ongoingChallengesNone.set('status', 'success'),
+        });
+        expect(wrapper.find('.withdraw-input').props().value).toEqual('0');
+      });
+    });
+    describe('confirm settling', () => {
+      [
+        { status: 'requesting', text: /waiting_for_confirm_settle_to_be/i },
+        { status: 'mining', text: /waiting_for_confirm_settle_to_be/i },
+        { status: 'receipt', text: /waiting_for_confirm_settle_to_be/i },
+      ].forEach((t) => {
+        it(`should hide the confirm button and show the transaction status when status is ${t.status}`, () => {
+          const wrapper = shallow(
+            <NahmiiWithdraw
+              {...props}
+              settleableChallenges={settleableChallengesNone.set('details', [
+                { intendedStageAmount: new nahmii.MonetaryAmount('100', '0x1', 0) },
+                { intendedStageAmount: new nahmii.MonetaryAmount('100', '0x1', 0) },
+              ]).set('status', t.status)}
+            />
+          );
+          const noticeNode = wrapper.find('style__SettlementWarning');
+          expect(noticeNode).toHaveLength(1);
+          expect(noticeNode.props().message).toEqual('settlement_period_ended');
+
+          const noticeNodeShallow = shallow(noticeNode.props().description);
+          expect(noticeNodeShallow.find('.confirm-btn')).toHaveLength(0);
```

Comment:
Is there a way to locate these DOM nodes without explicitly defining a className? Have you tried using the className assigned by the `styled-components` library?